### DISCUSSION TITLE
Open-source CPU targets

### DISCUSSION THREAD

**Discussion by taqtiqa-mark** [risc0/risc0#2323] [Q&A] [answered]
### Is there an existing issue?

- [X] I have searched the existing issues

### Experiencing problems? Have you tried our Discord first?

- [X] This is not a support question.

### Motivation

A desire to run ZKVMs on FPGA hardware.

### Request

Are any of the following open-source CPUs targets for Risc0?

### Solution

Specifically, should I ever expect that Risc0 will run on, say, the Berkeley/SiFive Rocket [rv32ima](https://github.com/chipsalliance/rocket-tools/blob/master/build-rv32ima.sh), etc

- [ ] Berkeley/SiFive Rocket - https://github.com/chipsalliance/rocket-chip
- [ ] Bluespec Flute - https://github.com/bluespec/Flute
- [ ] OpenHardware Group CVA6 - https://github.com/openhwgroup/cva6.git (a.k.a. ETH Zurich Ariane)
- [ ] PicoRV32 - https://github.com/YosysHQ/picorv32/tree/main
    - [RV32im instructions](https://github.com/YosysHQ/picorv32/tree/main#building-a-pure-rv32i-toolchain)

### Are you willing to help with this request?

Maybe... (I have elaborated on how above)

**Comment by linear** [comment]
<p><a href="https://linear.app/risczero/issue/ZKVM-526/open-source-cpu-targets">ZKVM-526 Open-source CPU targets</a></p>

**Comment by nuke-web3** [comment, marked as answer]
We are deprecating discussions on Github, and encouraging everyone to [join us on discord](https://discord.com/invite/risczero) where we can chat and provide support.

### [Join our Community Discord for dev chit-chat & our support forum](https://discord.com/invite/risczero)

For general questions and problems you are facing with RISC Zero, please **search** our `🙋｜dev-questions-support-forum` first for solutions, by keyword and tags. If you discover no solution, please then ask and questions in our community! We highly encourage everyone also share their understanding by answering questions for others!

Please do introduce yourself on our discord and send me a friend request (`@nuke_web3`) reminding me of this request so we can open up a forum post on it if there isn't a related one already there :grin:
